### PR TITLE
Android tutorial

### PR DESCRIPTION
**Pull request by mvpeterson**
Basic template for a demo app, contains: an ImageView and two buttons ("Open Image" and "Call Segmentation Service"). It uploads and shows an image from a device local storage.

The SNETServiceHelper class should do all the work on channel opening, service calling, and channel closing.

Requested changes from https://github.com/mvpeterson/img-segm-demo-app/pull/1 included here

### CODE REVIEW

File: example/android-tutorial/img-segm-demo-app/app/src/main/java/com/example/imagesegmentationdemo/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        progressBar.setVisibility(View.VISIBLE);
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE,
+                WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
+    }
+    public void setImageBitmap(Bitmap bm)
+    {
+        imageView.setImageBitmap(bm);
+    }
+
+    private List<String> getRequiredPermissions()
```
Comment (vsbogd) [inline example/android-tutorial/img-segm-demo-app/app/src/main/java/com/example/imagesegmentationdemo/MainActivity.java:119]: I would inline this method to have all permissions related code in one code block.

File: example/android-tutorial/img-segm-demo-app/app/src/main/java/com/example/imagesegmentationdemo/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+    private String[] appPermissions;
+
+    private static final int REQUEST_CODE_OPEN_IMAGE = 0;
+    private static final int REQUEST_CODE_PERMISSIONS = 1234;
+
+    public final int maxImageHeight = 1024;
+    public final int maxImageWidth = 1024;
+
+    public static AlertDialog.Builder newAlertDialogBuilder(Context context) {
```
Comment (vsbogd) [inline example/android-tutorial/img-segm-demo-app/app/src/main/java/com/example/imagesegmentationdemo/MainActivity.java:52]: Used only once can be inlined.

File: example/android-tutorial/img-segm-demo-app/app/src/main/java/com/example/imagesegmentationdemo/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+                .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int which)
+                    {
+                        dialog.dismiss();
+                    }
+                })
+                .setIcon(android.R.drawable.ic_dialog_alert);
+    }
+
+    public void msgShowException(Exception e)
```
Comment (vsbogd) [inline example/android-tutorial/img-segm-demo-app/app/src/main/java/com/example/imagesegmentationdemo/MainActivity.java:64]: Not used can be removed

File: example/android-tutorial/img-segm-demo-app/app/src/main/java/com/example/imagesegmentationdemo/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+    private Uri imageUri;
+
+    private String[] appPermissions;
+
+    private static final int REQUEST_CODE_OPEN_IMAGE = 0;
+    private static final int REQUEST_CODE_PERMISSIONS = 1234;
+
+    public final int maxImageHeight = 1024;
+    public final int maxImageWidth = 1024;
```
[the remark this answers is not in the document]
Comment (vsbogd) [reply example/android-tutorial/img-segm-demo-app/app/src/main/java/com/example/imagesegmentationdemo/MainActivity.java:50]: And these constants could be moved into `SNETServiceHelperClass` the only place where values are used.

File: example/android-tutorial/img-segm-demo-app/app/build.gradle [hunk 5]
```diff
@@ -0,0 +1,56 @@
+apply plugin: 'com.android.application'
+apply plugin: 'io.singularitynet.sdk'
```
Comment (vsbogd) [inline example/android-tutorial/img-segm-demo-app/app/build.gradle:2]: Plugin should not be needed here.
```suggestion
```

File: example/android-tutorial/img-segm-demo-app/semantic-segmentation-api/build.gradle [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        sourceCompatibility = 1.8
+        targetCompatibility = 1.8
+    }
+}
+
+protobuf {
+    protoc { artifact = "com.google.protobuf:protoc:3.5.1" }
+    plugins {
+        java
+        grpc { artifact = "io.grpc:protoc-gen-grpc-java:1.20.0" }
```
Comment (vsbogd) [inline example/android-tutorial/img-segm-demo-app/semantic-segmentation-api/build.gradle:72]: I am not 100% but probably here should be ${grpcVersion}
```suggestion
        grpc { artifact = "io.grpc:protoc-gen-grpc-java:${grpcVersion}" }
```

File: example/android-tutorial/img-segm-demo-app/semantic-segmentation-api/build.gradle [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        }
+    }
+    compileOptions {
+        sourceCompatibility = 1.8
+        targetCompatibility = 1.8
+    }
+}
+
+protobuf {
+    protoc { artifact = "com.google.protobuf:protoc:3.5.1" }
```
Comment (vsbogd) [inline example/android-tutorial/img-segm-demo-app/semantic-segmentation-api/build.gradle:69]:
```suggestion
    protoc { artifact = "com.google.protobuf:protoc:${protobufVersion}" }
```

File: example/android-tutorial/img-segm-demo-app/semantic-segmentation-api/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.example.semantic_segmentation_api">
```
Comment (vsbogd) [inline example/android-tutorial/img-segm-demo-app/semantic-segmentation-api/src/main/AndroidManifest.xml:2]: I would suggest `io.singularitynet.sdk.tutorial` package as well

**Review by vsbogd — CHANGES_REQUESTED** [review]

File: example/android-tutorial/img-segm-demo-app/app/build.gradle [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+android {
+    compileSdkVersion 29
+
+    defaultConfig {
+        applicationId "com.example.imagesegmentationdemo"
```
[the remark this answers is not in the document]
Comment (mvpeterson) [reply example/android-tutorial/img-segm-demo-app/app/build.gradle]: fixed in https://github.com/singnet/snet-sdk-java/pull/62/commits/79a2cf2865d96896b04868fe4f408407e9f969c9

File: example/android-tutorial/img-segm-demo-app/LICENSE [hunk 10]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2020 Maxim Peterson
```
[the remark this answers is not in the document]
Comment (mvpeterson) [reply example/android-tutorial/img-segm-demo-app/LICENSE]: fixed in https://github.com/singnet/snet-sdk-java/pull/62/commits/11a4382f7a0a1c843c1e409337a2049bb95bbc0d

File: example/android-tutorial/img-segm-demo-app/app/src/main/java/com/example/imagesegmentationdemo/ImageUtils.java [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+    /**
+     * This method is responsible for solving the rotation issue if exist. Also scale the images to
+     * maxImageWidth x maxImageHeight resolution
+     *
+     * @param context       The current context
+     * @param selectedImage The Image URI
+     * @return Bitmap image results
+     * @throws IOException
+     */
+    public static Bitmap handleSamplingAndRotationBitmap(Context context, Uri selectedImage,
```
[the remark this answers is not in the document]
Comment (mvpeterson) [reply example/android-tutorial/img-segm-demo-app/app/src/main/java/com/example/imagesegmentationdemo/ImageUtils.java]: applied in https://github.com/singnet/snet-sdk-java/pull/62/commits/4cd00a083aead2352b18c7db06449ff8346fb27f

File: example/android-tutorial/img-segm-demo-app/semantic-segmentation-api/src/main/AndroidManifest.xml (hunk 8 above)
Comment (mvpeterson) [reply example/android-tutorial/img-segm-demo-app/semantic-segmentation-api/src/main/AndroidManifest.xml:2]: applied in https://github.com/singnet/snet-sdk-java/pull/62/commits/79a2cf2865d96896b04868fe4f408407e9f969c9

Comment (vsbogd) [reply example/android-tutorial/img-segm-demo-app/semantic-segmentation-api/src/main/AndroidManifest.xml:2]: Here the package is still `com.example`

File: example/android-tutorial/img-segm-demo-app/app/build.gradle [hunk 12]
```diff
@@ -0,0 +1,55 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion 29
+
+    defaultConfig {
+        applicationId "io.singularitynet.sdk.tutorial"
```
Comment (vsbogd) [inline example/android-tutorial/img-segm-demo-app/app/build.gradle:7]:
```suggestion
        applicationId "io.singularitynet.sdk.tutorial.semanticsegmentation"
```

File: example/android-tutorial/img-segm-demo-app/semantic-segmentation-api/src/main/AndroidManifest.xml (hunk 8 above)
Comment (mvpeterson) [reply example/android-tutorial/img-segm-demo-app/semantic-segmentation-api/src/main/AndroidManifest.xml]: fix is in https://github.com/singnet/snet-sdk-java/pull/62/commits/60108d6f1894fbfce96e03bcba853eda11400bfe

**Review by vsbogd — APPROVED** [review]

**Pull request merged**